File: backend/src/main/java/com/nfcplatform/menu/entity/Menu.java
package com.nfcplatform.menu.entity;

import com.nfcplatform.common.entity.Auditable;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "menus")
public class Menu extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, updatable = false, length = 36)
    private String uuid = UUID.randomUUID().toString();

    @Column(name = "client_id", nullable = false, unique = true)
    private Long clientId;

    @Column(nullable = false, length = 100)
    private String slug;

    @Column(nullable = false)
    private String name;

    private String logo;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false, length = 10)
    private String currency = "LKR";

    @Column(nullable = false)
    private boolean published = false;
}

File: frontend/src/lib/types/menu.ts
export interface MenuItem {
  uuid: string;
  name: string;
  description: string | null;
  image: string | null;
  price: number;
  available: boolean;
  featured: boolean;
  sortOrder: number;
}

export interface MenuCategory {
  uuid: string;
  name: string;
  sortOrder: number;
  active: boolean;
  items: MenuItem[];
}

export interface Menu {
  uuid: string;
  slug: string;
  name: string;
  description: string | null;
  logo: string | null;
  currency: string;
  published: boolean;
  publicUrl: string;
  categories: MenuCategory[];
}
